Sources/MapboxCoreNavigation/EventDetails.swift

```diff
@@ -136,6 +137,10 @@ struct NavigationEventDetails: EventDetails {
         sessionIdentifier = session.identifier.uuidString
         originalRequestIdentifier = session.originalRoute.routeIdentifier
         requestIdentifier = dataSource.routeProgress.route.routeIdentifier
+        
+        if (userInfo != nil) {
```

Comment:
Do we really need this check? Both `userInfo` and `appMetadata` are Optional, you can just assign it directly.